### PR TITLE
Implement IsOpndNegZero in the JIT

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

**Comment by meg-gupta** [comment]
WIP

**Comment by Cellule** [comment]
Related to #2041 
Let me know when it's ready to review

**Comment by meg-gupta** [comment]
@Cellule Enabled for x86 as well, ready for review.

File: lib/Backend/amd64/LowererMDArch.cpp [hunk 1]
```diff
@@ -3255,3 +3255,62 @@ LowererMDArch::LowerEHRegionReturn(IR::Instr * insertBeforeInstr, IR::Opnd * tar
...
+    this->m_func->StackAllocate(symOpnd->m_sym->AsStackSym(), sizeof(double));
+
+    // LEA r3, stackSym
+    IR::RegOpnd *addrOpnd = IR::RegOpnd::New(TyMachReg, this->m_func);
+    IR::Instr *lea = IR::Instr::New(Js::OpCode::LEA, addrOpnd, symOpnd, this->m_func);
+    instr->InsertBefore(lea);
+
+    // MOVSD [stackSym], src
+    IR::IndirOpnd *indirOpnd = IR::IndirOpnd::New(addrOpnd, 0, TyFloat64, this->m_func);
+    IR::Instr *movsd = IR::Instr::New(Js::OpCode::MOVSD, indirOpnd, src, this->m_func);
```
Comment (meg-gupta) [inline lib/Backend/amd64/LowererMDArch.cpp]: Thanks @MikeHolman , updated the diff. x86 still requires the lengthy lowering. I saw that SSE4 has PEXTRD which will help generate better code for x86, but I haven't updated the code for this yet.

**Comment by MikeHolman** [comment]
:shipit:

**Pull request merged**